algorithms.py

```diff
@@ -700,6 +1022,18 @@ def change_population_growth_rate(self, pop_id, rate, time):
     def change_migration_matrix_element(self, pop_i, pop_j, rate):
         self.migration_matrix[pop_i][pop_j] = rate
 
+    def alloc_hull(self, left, right, ancestor_node):
```

Comment:
I don't get why this is called ``ancestor_node``. This must be a segment or something, if we can ``prev`` it?

Reply:
I think we need a different name here all right, this is confusing me quite a bit